pennylane_lightning/src/rework/GateFactory.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright 2021 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "GateFactory.hpp"
+
+using std::string;
+using std::unique_ptr;
+using std::vector;
+
+// FIXME: This should be reworked to use a function dispatch table
```

Comment:
> function dispatch table

Would that be something like `std::map<std::string, func> ` where `func` is the function that returns the gate of the operator? (Something like [what is there for the current ops](https://github.com/PennyLaneAI/pennylane-lightning/blob/dd9efc956e6f7bfd4fe5293a28842b9603f66521/pennylane_lightning/src/operations.hpp#L359)).